tools/build/Makefile.vars

```diff
@@ -282,7 +282,7 @@ COMMON_CFLAGS += -fPIC
 endif
 
 ifeq (y,$(BUILD_TYPE_DEBUG))
-COMMON_CFLAGS += -g -fno-lto
+COMMON_CFLAGS += -g -fno-lto -O0
 endif
```

Comment:
On Fri, 01 Apr 2016 07:07:28 -0700, Flávio Vinicius Alvares Ceolin wrote:
> > @@ -282,7 +282,7 @@ COMMON_CFLAGS += -fPIC
> >  endif
> >  
> >  ifeq (y,$(BUILD_TYPE_DEBUG))
> > -COMMON_CFLAGS += -g -fno-lto
> > +COMMON_CFLAGS += -g -fno-lto -O0
> >  endif
> 
> o yeah, here is much better

I'd rather re-order the way we add stuff to COMMON_CFLAGS to make sure
that all the things that can break debugging are not there in the first
place.